src/panel/direction/DirectionPanel.jsx

```diff
@@ -222,18 +223,18 @@ export default class DirectionPanel extends React.Component {
   }
 
   updateUrl() {
-    const routeParams = [];
-    if (this.state.origin) {
-      routeParams.push('origin=' + poiToUrl(this.state.origin));
-    }
-    if (this.state.destination) {
-      routeParams.push('destination=' + poiToUrl(this.state.destination));
-    }
-    routeParams.push(`mode=${this.state.vehicle}`);
-    if (this.props.isPublicTransportActive) {
-      routeParams.push('pt=true');
-    }
-    window.app.navigateTo(`/routes/?${routeParams.join('&')}`, {}, {
+    const queryObject = {
+      ...parseQueryString(window.location.search),
+      mode: this.state.vehicle,
+      origin: this.state.origin ? poiToUrl(this.state.origin) : null,
+      destination: this.state.destination ? poiToUrl(this.state.destination) : null,
+      pt: this.props.isPublicTransportActive ? 'true' : null,
+    };
+
+    const search = buildQueryString(queryObject);
+    const relativeUrl = buildRelativeUrl({ search });
+
+    window.app.navigateTo(relativeUrl, {}, {
```

Comment:
`buildRelativeUrl` uses the current pathname, possibly prefixed with `/maps/`, while `navigateTo` does not support this prefix and expects the application route only (eg `/routes` without prefix)

Reply:
I think we can do without `buildRelativeUrl`. It is ok for the component to now what hardcoded route it needs.